packages/core/src/redocly/index.ts

```diff
@@ -1,58 +1,78 @@
 import { existsSync, readFileSync, writeFileSync, unlinkSync } from 'fs';
 import { resolve } from 'path';
 import { homedir } from 'os';
-import { yellow, red, green, gray } from 'colorette';
+import { red, green, gray } from 'colorette';
 import { RegistryApi } from './registry-api';
+import { DOMAINS, Region } from '../config/config';
 
 const TOKEN_FILENAME = '.redocly-config.json';
+const DEFAULT_REGION = 'us';
+const DEFAULT_DOMAIN = 'redoc.ly';
 
 export class RedoclyClient {
-  private accessToken: string | undefined;
+  private accessTokens: { us?: string; eu?: string; } | undefined;
+  private domain: string | undefined;
+  private region: Region;
   registryApi: RegistryApi;
 
-  constructor() {
-    this.loadToken();
-    this.registryApi = new RegistryApi(this.accessToken);
+  constructor(region?: Region) {
+    this.region = region || DEFAULT_REGION;
+    this.loadTokens();
+    this.setDomain(region);
+    this.registryApi = new RegistryApi(this.getTokenByRegion(), this.domain);
   }
 
-  hasToken(): boolean {
-    return !!this.accessToken;
+  getTokenByRegion() {
+    return this.accessTokens && this.accessTokens[this.region];
   }
 
-  loadToken(): void {
+  setDomain(region?: Region) {
+    this.domain = region
+      ? DOMAINS[region]
+      : process.env.REDOCLY_DOMAIN || DOMAINS[DEFAULT_REGION];
+  }
+
+  getDomain() {
+    return this.domain || DEFAULT_DOMAIN;
+  }
+
+  hasTokens(): boolean {
+    return !!this.accessTokens;
+  }
+
+  loadTokens(): void {
     if (process.env.REDOCLY_AUTHORIZATION) {
-      this.accessToken = process.env.REDOCLY_AUTHORIZATION;
+      this.accessTokens = {};
+      for (const domain in DOMAINS) {
+        this.accessTokens[domain as Region] = process.env.REDOCLY_AUTHORIZATION;
+      }
       return;
     }
-
     const credentialsPath = resolve(homedir(), TOKEN_FILENAME);
     if (existsSync(credentialsPath)) {
-      const credentials = JSON.parse(readFileSync(credentialsPath, 'utf-8'));
-      this.accessToken = credentials && credentials.token;
+      this.accessTokens = JSON.parse(readFileSync(credentialsPath, 'utf-8'));
```

Comment:
This is backward-incompatible.

Most of the users have the following creds content on their machines:

```
{
  token: '....'
}
```

Those tokens will stop working. We need to have some logic to map old format to new.

I think if you move whole `JSON.parse(readFileSync(credentialsPath, 'utf-8'));` to separate function you can map it there.

I think it should do something like that:

```
const tokens = JSON.parse(readFileSync(credentialsPath, 'utf-8'));
if (tokens.token && !tokens[this.domain]) tokens[domain] = tokens.token;
```